### PR TITLE
Cleanup warnings

### PR DESCRIPTION
**Pull request by wvlothuizen**

### CODE REVIEW

File: generator/tree-gen-cpp.cpp [hunk 1]
```diff
@@ -505,7 +505,7 @@ void generate_node_class(
                 source << "    node->" << field.name << " = this->" << field.name << ".clone();" << std::endl;
             }
         }
-        source << "    return node;" << std::endl;
+        source << "    return std::move(node);" << std::endl;
```
Comment (jvanstraten) [inline generator/tree-gen-cpp.cpp]: Does this actually fix anything? It is my understanding that explicitly moving a return value is very rarely if ever a good idea. There are already exceptions in the spec that ensure that a return value will be moved in most cases even if it is an lvalue, and explicitly wrapping something in `std::move` will prevent regular return value optimization/elision if the function is inlined. See for example https://diego.assencio.com/?index=f57f25fd5a187c70fc7f34fcf5374773

**Comment by wvlothuizen** [comment]
@jvanstraten could you review this PR

**Review by jvanstraten — CHANGES_REQUESTED** [review]

**Comment by jvanstraten** [comment]
Guess I forgot to push the "submit review" button 9 days ago.....

File: generator/tree-gen-cpp.cpp (hunk 1 above)
Comment (wvlothuizen) [reply generator/tree-gen-cpp.cpp]: Oops, I must confess that I was lead by the compiler issuing warnings (a lot of them), but reading this it my change worsens the code. Learning all the time...
Shall I change the PR, leaving only the typo, or shall we just cancel it?

Comment (jvanstraten) [reply generator/tree-gen-cpp.cpp]: Up to you.

**Comment by wvlothuizen** [comment]
Well, that leaves a single character change in this PR

**Review by jvanstraten — APPROVED** [review]

**Pull request merged**